emulator/src/axi4_slave.hpp

```diff
@@ -0,0 +1,269 @@
+#ifndef AXI4_SLAVE
+#define AXI4_SLAVE
+
+#include "axi4.hpp"
+
+#include <queue>
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+template <unsigned int A_WIDTH = 64, unsigned int D_WIDTH = 64, unsigned int ID_WIDTH = 4>
+class axi4_slave {
+    static_assert(D_WIDTH <= 64, "D_WIDTH should be <= 64.");
```

Comment:
can it be larger than 64? since for some in-chip bus, 128/256 width is pretty common.